avm/res/storage/storage-account/main.bicep

```diff
@@ -69,10 +69,7 @@ param privateEndpoints privateEndpointType
 param managementPolicyRules array?
 
 @description('Required. Networks ACLs, this value contains IPs to whitelist and/or Subnet information. If in use, bypass needs to be supplied. For security reasons, it is recommended to set the DefaultAction Deny.')
-param networkAcls networkAclsType = {
-  bypass: 'AzureServices'
-  defaultAction: 'Deny'
-}
+param networkAcls networkAclsType
```

Comment:
Why removing the default? Would it not make more sense to move it into the network ACLS like suggested here: https://github.com/Azure/bicep-registry-modules/pull/1472#discussion_r1546120751

Reply:
Just wondering how we could make it so that a firewall would be enabled by default, but could be disabled if needed. I guess with the implementation in the comment it's also not exactly possible, right? 😄

Reply:
Somehow in my mind it doesn't check out...

Reply:
@eriqua / @ChrisSidebotham any ideas?

Reply:
Removing the default value was the only solution I was able to come up with. Everything else just fails the linting spectacularly, maybe you guys have a better idea on how to achieve this?

Reply:
I just checked something again. Although the linter goes absolutely crazy, it would still work if we use `Action: Allow`? So the default would still be `Action: Deny` but a user could just specify a networkACLType with `Action: Allow` in it. Right?

Reply:
I had the same in [my PR](https://github.com/Azure/bicep-registry-modules/pull/1524/files), but I figured it's probably better to remove the `?` after the definition of the `networkAclsType`.

Reply:
In [my PR](https://github.com/Azure/bicep-registry-modules/pull/1524) I was thinking of making the `networkAclsType` not nullable. Wouldn't that be an option? Or should users also be able to actually set this value to `null`?

Reply:
Hey @bartcode,
the challenge comes when a user should be able to not have the firewall enabled. If the value isn't nullable (or has an empty default of sorts), the user must provide it. As per the UDT, the user can only provide the value if 2 required properties are set as well. However, in either case this enables the firewall.
Now, if the parameter is nullable, one could in theory avoid to set the value. However, today, this still leads to the firewall being enabled because there is a default.
Hence @fblix's solution of not setting a default works. But of course there is a but: You'd actually want to enable the firewall by default to ensure the storage account is secured by default. It may even be required to satisfy PSRule's WAF rules.
So ... it's a bit of a mess :D 

An alternative would be to have the UDT nullable, and not set a default in the parameter. Instead, the place where the parameter is used could differentiate 'hey is the value null or not. If yes, I enable the firewall, of not I set the configuration'. And a user could deliberatly provide an empty object for the parameter to disable the firewall. The only challenge with that (today) is, that the UDT enforces 2 of the properties of set object. So you cannot actually provide an empty object as an input.

fun fun fun

Reply:
Thanks for the detailed explanation! I think I'll just wait until this PR is merged and then I'll make my - rather small - change.

Reply:
> @eriqua / @ChrisSidebotham any ideas?

Still need to look into this more closely.

The rule of thumb, if that helps, is to make it more difficult, rather than easier, to deviate from the value we want to enforce as default. That is only in case the intention of the default is to make the resource secure by default/WAF compliant/pass PSRule.

Regarding the implementation, please consider the condition for public network access, normally enabled or disabled by default also based on network ACL values.

Reply:
Hey @fblix,
I took another stab.

How about this to avoid making it mandatory.

Template
```bicep
@description('Required. Networks ACLs, this value contains IPs to whitelist and/or Subnet information. If in use, bypass needs to be supplied. For security reasons, it is recommended to set the DefaultAction Deny.')
param networkAcls networkAclsType

(...)
    networkAcls: !empty(networkAcls)
      ? {
          resourceAccessRules: networkAcls.?resourceAccessRules
          bypass: networkAcls.?bypass
          defaultAction: networkAcls.?defaultAction ?? 'Deny'
          virtualNetworkRules: networkAcls.?virtualNetworkRules
          ipRules: networkAcls.?ipRules
        }
      : {
          bypass: 'AzureServices'
          defaultAction: 'Deny'
        }

type networkAclsType = {
  (...)  
}?
(...)
``` 

Test case (defaults test)
```bicep
module testDeployment '../../../main.bicep' = [
  for iteration in ['init', 'idem']: {
    scope: resourceGroup
    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
    params: {
      name: '${namePrefix}${serviceShort}001'
      allowBlobPublicAccess: false
      location: resourceLocation
      networkAcls: {
        defaultAction: 'Allow'
        bypass: 'None'
      }
    }
  }
]
``` 

In Azure, that showed up as 
![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/02f8b34b-acd5-489e-88f6-bf891e3e42ec)

So the firewall was disabled.

When I remove the explicit specification / it falls back to the default it shows
![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/1cd5ba27-d85c-4f13-b7ec-752530bab6a7)

Reply:
+1 on @AlexanderSehr proposed implementation. 

Not sure I follow on the defaults test case though. 
Shouldn't we just use required parameters there? I think I'm missing something.

Reply:
You're correct @eriqua. I just used the default test as a showcase to not muddy the waters.